apps/verticalhome/js/pin_apps.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+(function (exports) {
+
+  function PinAppItem(index) {
+    if (index === null || index === undefined){
+      return false;
+    }
+    this.index = index;
+    //creating pin app element
+    var pinElem = document.createElement('div');
+    pinElem.className = 'pin-app-item';
+    pinElem.setAttribute('data-index', this.index);
+    pinElem.innerHTML = "<img class='pin-app-icon'><br></img>" +
+            "<span class='title'></span>";
+    var pinList = document.getElementById('pin-apps-list');
+    var moreAppsLi = document.getElementById('moreApps');
+    pinList.insertBefore(pinElem, moreAppsLi);
+
+    this.element = pinElem;
+    this.icon = this.element.querySelector('.pin-app-icon');
+    this.title = this.element.querySelector('.title');
+  }
+
+  PinAppItem.prototype = {
+    entry: null,
+
+    bindEntry: function(entry) {
```

Comment:
could you combine this with the constructor and pass the entry object to the constructor instead of having this separate method?  Also should entry be called something more specific like manifest?